yrs/src/block_store.rs

```diff
@@ -72,8 +73,11 @@ impl ClientBlockList {
         if self.integrated_len == 0 {
             0
         } else {
-            let item = &self.list[self.integrated_len - 1];
-            item.id.clock + 1
+            match &self.list[self.integrated_len - 1] {
+                Block::Item(item) => item.id.clock + 1,
+                Block::Skip(skip) => skip.len + 1,
+                Block::GC(gc) => gc.len + 1,
```

Comment:
I used intuition here, based on how `Store::from` works. I have no other reason to think this is correct.

Reply:
The state of a client is basically the next expected id from a certain client. 

So assume you apply item `{ id: { client: 0, clock: 0 }, len: 1 }` and then  `{ id: { client: 0, clock: 1 }, len: 2 }`. The next expected item from client 0 is ` { id: { client: 0, clock: 3 }, len: _ }`. 

So after applying the first operation, the state is `1`. After applying the next operation (which has clock = 1), the state is 3. After applying the third operation, the state is 3 + len (depending on the length of the operation). 

The length of an operation is determined by the amount of content an item fits. I use a compound representation to represent several insertions in a single item.

We should talk about the idea of compound representation of operations. I talked about my motivation behind them in my blog: https://blog.kevinjahns.de/are-crdts-suitable-for-shared-editing/

This should be `item.id.clock + item.len()`, and `skip.id.clock + skip.len()`, and `gc.id.clock + gc.len()` respectively.